### PR TITLE
ddl: add require owner interface

### PR DESCRIPTION
**Pull request by xiongjiwei**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/32031

Problem Summary:

### What is changed and how it works?

add `RequireOwner` into `OwnerManager` interface
change `DDLOwnerChecker` to `IsDDLOwner` in `sessionctx/context.go`
move `owner.NewSession` to util package to avoid cycle import

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by xiongjiwei** [comment]
@hawkingrei @wjhuang2016 PTAL

**Review by wjhuang2016 — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

File: ddl/util/util.go [hunk 1]
```diff
@@ -204,6 +212,78 @@ func LoadGlobalVars(ctx context.Context, sctx sessionctx.Context, varNames []str
...
+	logIntervalCnt          = int(3 * time.Second / newSessionRetryInterval)
+
+	// NewSessionDefaultRetryCnt is the default retry times when create new session.
+	NewSessionDefaultRetryCnt = 3
+	// NewSessionRetryUnlimited is the unlimited retry times when create new session.
+	NewSessionRetryUnlimited = math.MaxInt64
+)
+
+// NewSession creates a new etcd session.
+func NewSession(ctx context.Context, logPrefix string, etcdCli *clientv3.Client, retryCnt, ttl int) (*concurrency.Session, error) {
```
Comment (zimulala) [inline ddl/util/util.go:226]: Why did we put this function here? This function is not necessarily related to DDL, as it is also used in "info.go"?

Comment (zimulala) [inline ddl/util/util.go:226]: Why did we put this function here? This function is not necessarily related to DDL, as it is also used in "info.go"?

**Review by zimulala — CHANGES_REQUESTED** [review]

File: ddl/util/util.go (hunk 1 above)
Comment (xiongjiwei) [reply ddl/util/util.go:226]: well, `tidb/util` is a better place I think?

Comment (zimulala) [reply ddl/util/util.go:226]: If you want to put it on `util`, I think so. I'm not sure why we need to remove it from `owner`. You can also describe this in PR

Comment (xiongjiwei) [reply ddl/util/util.go:226]: it will be cycle import after implementing `requireOwner`.

Comment (zimulala) [reply ddl/util/util.go:226]: Got it. It can be explained in PR, otherwise, others may not be clear about it.

**Comment by xiongjiwei** [comment]
@zimulala PTAL

**Review by zimulala — APPROVED** [review]
LGTM

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**